x-pack/plugins/license_api_guard/server/license.ts

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { i18n } from '@kbn/i18n';
+import {
+  Logger,
+  KibanaRequest,
+  KibanaResponseFactory,
+  RequestHandler,
+  RequestHandlerContext,
+} from 'src/core/server';
+
+import { ILicense, LicenseType, LicenseCheckState, LicensingPluginStart } from './shared_imports';
+
+type LicenseLogger = Pick<Logger, 'warn'>;
+type LicenseDependency = Pick<LicensingPluginStart, 'license$'>;
+
+interface SetupSettings {
+  pluginName: string;
+  logger: LicenseLogger;
+}
+
+interface StartSettings {
+  pluginId: string;
+  minimumLicenseType: LicenseType;
+  licensing: LicenseDependency;
+}
+
+export class License {
+  private pluginName?: string;
+  private logger?: LicenseLogger;
+  private licenseCheckState: LicenseCheckState = 'unavailable';
+  private licenseType?: LicenseType;
+
+  private _isEsSecurityEnabled: boolean = false;
+
+  setup({ pluginName, logger }: SetupSettings) {
+    this.pluginName = pluginName;
+    this.logger = logger;
+  }
+
+  start({ pluginId, minimumLicenseType, licensing }: StartSettings) {
+    licensing.license$.subscribe((license: ILicense) => {
```

Comment:
shouldn't we unsubscribe on the `stop` lifecycle?

Reply:
@mshustov I was under the impression that `stop` is never called in server-side plugins -- plugins just "live" until the Kibana server process is killed. Is that not the case?

Reply:
> plugins just "live" until the Kibana server process is killed.

Nope. It's always called during the normal teardown process (unless Kibana crashes).

Reply:
When does teardown occur? How can I trigger it?